sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/entity_resolution/Commons.scala

```diff
@@ -16,17 +15,22 @@ import org.apache.spark.RangePartitioner
 import org.apache.spark.storage.StorageLevel
 import org.apache.spark.RangePartitioner
 import org.apache.spark.RangePartitioner
+import scala.io.Source
+import java.io.Serializable
 
 /* A generic Entity Resolution Approach for larger datasets(for e.g. 24.8GB)
  * Dataset1 has 47 million entities, approx 19GB
  * Dataset2 has 19 million entities, approx 5.8GB
  * */
-class EntityResolution_RDFData_CountVectorizer(spark: SparkSession, triplesRDD1: RDD[Triple], triplesRDD2: RDD[Triple],
-                                         teacher: DataFrame, threshold_subject: Double, jsimilarity_predicate: Double,
-                                         threshold_object: Double, vocab_size: Long, output_path: String) extends Serializable {
+abstract class Commons(val spark: SparkSession, val sourceData1: RDD[Triple], val sourceData2: RDD[Triple],
```

Comment:
Why `Commons`? and why `abstract`? To my understanding, _The abstract methods of an abstract class are those methods that do not contain any implementation._ and in your case, your methods are fully implemented. I will suggest that you create a package `utils` which then contains such methods (grouped by their functionality into separate Scala `objects`).

Reply:
The Commons class contains the abstract method vectorise which has seperate implemntations in the two subclass that inherit this class. That's why it is structured in this way. If you still think I should do it in some other better way. PLease suggest.

Reply:
Not sure, but feel free to keep it as you have if you think that is the right way to do. Later, we can refactor it if needed.